python/Ganga/Utility/execute.py

```diff
@@ -184,24 +251,46 @@ def execute(command,
     if timed_out.isSet():
         return 'Command timed out!'
 
-    if update_env and 'env_output' in thread_output:
-        env.update(thread_output['env_output'])
+    # Decode any pickled objects from disk
+    if update_env:
+        update_env_thread.join()
+        if env_output_key in thread_output:
+            env.update(thread_output[env_output_key])
+        else:
+            logger.error("Expected to find the updated env after running a command")
+            logger.error("Command: %s" % command)
+            logger.error("stdout: %s" % stdout)
+            logger.error("stderr: %s" % stderr)
+            raise Exception("Missing update env after running command")
```

Comment:
This (and the one below) should be more specific in their exception type. I would suggest at least `RuntimeError`.